modules/tinymce/src/core/main/ts/keyboard/EnterKey.ts

```diff
@@ -21,7 +21,11 @@ const handleEnterKeyEvent = (editor: Editor, event: EditorEvent<KeyboardEvent>)
   endTypingLevelIgnoreLocks(editor.undoManager);
   editor.undoManager.transact(() => {
     if (editor.selection.isCollapsed() === false) {
+      // Hacky way to avoid input events to be fired by execCommand idealy we would have our own delete implementation
```

Comment:
Not sure about this change. Delete and range.deleteContents() is not doing the same thing for empty blocks so they are not 1:1 compatible and we don't have our own custom delete logic that works in all cases. So in order to prevent our override to also fire a input event for execCommand it put it into readonly mode that blocks the input events from firing by the event delegation but it feels a bit hacky. But the alternative would be to add some form of flag separate from readonly to only deal with this specific situation to cancel events. This was the best I could come up with.